src/api/resetIndex.js

```diff
@@ -26,38 +28,57 @@ import { resolveFilepath } from '../utils/resolveFilepath.js'
  * await git.resetIndex({ fs, dir: '/tutorial', filepath: 'README.md' })
  * console.log('done')
  *
+ * @example
+ * await git.resetIndex({ fs, dir: '/tutorial', filepath: 'README.md', oid: '11a57ba27f571e9e2bfa344fdbbccad1d0e34463' })
+ * console.log('done')
+ *
  */
 export async function resetIndex({
   fs: _fs,
   dir,
   gitdir = join(dir, '.git'),
   filepath,
-  ref = 'HEAD',
+  ref,
+  oid,
   cache = {},
 }) {
   try {
     assertParameter('fs', _fs)
     assertParameter('gitdir', gitdir)
     assertParameter('filepath', filepath)
-    assertParameter('ref', ref)
 
     const fs = new FileSystem(_fs)
-    // Resolve commit
-    let oid = await GitRefManager.resolve({ fs, gitdir, ref })
+
     let workdirOid
-    try {
-      // Resolve blob
-      oid = await resolveFilepath({
-        fs,
-        cache,
-        gitdir,
-        oid,
-        filepath,
-      })
-    } catch (e) {
-      // This means we're resetting the file to a "deleted" state
-      oid = null
+
+    if (!oid) {
+      try {
+        // Resolve ref if no oid is provided
+        oid = await GitRefManager.resolve({ fs, gitdir, ref: ref || 'HEAD' })
+      } catch (e) {
+        if (ref) {
+          // Only throw the error if a ref is explicitly provided
+          throw e
+        }
+      }
+    }
+
+    if (oid) {
```

Comment:
It only makes sense to resolve the file path of we have an `oid` at this point, if we don't have one we can skip it. The reason we don't have one is in the case of a `resetIndex()` without explicit `ref` on a new git repository.

Reply:
Please add a comment with that info. It may be useful to have this in code.

Reply:
Good idea! Added it.